### ISSUE TITLE
Why not remove outdated parts permanently after TTL merge?

### ISSUE THREAD

**Issue by kekekedeng** [ClickHouse/ClickHouse#16629]
##  Description 

I find some empty parts after background merge. These parts are old enough to be deleted permanently. Why does it still exists?

![image](https://user-images.githubusercontent.com/5371877/97964961-955c1e00-1df4-11eb-88ce-76cb0fc974cd.png)

![image](https://user-images.githubusercontent.com/5371877/97964803-4f9f5580-1df4-11eb-9c34-b6239713a6ef.png)

### Question1: Does it work as expect? 

I have read the source code about background merge logic, but not found logic to delete it.

### Question2: Can it cause "too many parts" problem?

Maybe, it can exceeds the parts amount limitation `max_parts_in_total`.

Now, i can avoid this problem by the following two steps:

* Wait for background merge or trigger merge manually by "OPTIMIZE xxx"  to ensure that only one part for one outdated partition.
* Drop outdated partition by "ALTER TABLE table_name DROP PARTITION XXX".

### Question3: Why not remote it permanently?

Why not do this? Are there any other considerations?

CH version 19.16.20.1.

**Comment by xjewer** [comment]
there's an umbrella ticket covering TTL issues https://github.com/ClickHouse/ClickHouse/issues/10128

tl;dr 19.16 version is quite old for it, consider an update.

**Comment by den-crane** [comment]
https://github.com/ClickHouse/ClickHouse/pull/13252#issuecomment-667734024
